presto-main/src/main/java/com/facebook/presto/testing/MaterializedResult.java

```diff
@@ -424,6 +425,51 @@ public static MaterializedResult materializeSourceDataStream(ConnectorSession se
         return builder.build();
     }
 
+    public static boolean resultsEqual(MaterializedResult viewTable, MaterializedResult baseTable)
+    {
+        assertEquals(viewTable.getRowCount(), baseTable.getRowCount());
+
+        Iterator<MaterializedRow> viewRowsIterator = viewTable.iterator();
+        Iterator<MaterializedRow> baseRowsIterator = baseTable.iterator();
+        List<Type> viewTypes = viewTable.getTypes();
+        List<Type> baseTypes = baseTable.getTypes();
+
+        while (viewRowsIterator.hasNext()) {
+            MaterializedRow viewRow = viewRowsIterator.next();
+            MaterializedRow baseRow = baseRowsIterator.next();
+
+            List<Object> viewFields = viewRow.getFields();
+            List<Object> baseFields = baseRow.getFields();
+            assertEquals(viewFields.size(), baseFields.size());
+
+            Iterator<Object> viewFieldIterator = viewFields.iterator();
+            Iterator<Object> baseFieldIterator = baseFields.iterator();
+
+            Iterator<Type> viewTypeIterator = viewTypes.iterator();
+            Iterator<Type> baseTypeIterator = baseTypes.iterator();
+
+            while (viewFieldIterator.hasNext()) {
+                Type type = viewTypeIterator.next();
+                assertEquals(type, baseTypeIterator.next());
+
+                Object viewField = viewFieldIterator.next();
+                Object baseField = baseFieldIterator.next();
+                if (type.isOrderable()) {
+                    if (((Comparable) viewField).compareTo(baseField) != 0) {
+                        return false;
+                    }
+                }
+                else if (type.isComparable()) {
+                    if (!viewField.equals(baseField)) {
+                        return false;
+                    }
+                }
+            }
+        }
+
+        return true;
```

Comment:
No need to return true or false. Directly throw if they are different.

Reply:
Can we confirm if we ran all tests by enabling them?